### ISSUE TITLE
revision of the STRENDA modelling principles

### ISSUE THREAD

**Issue by markdoerr** [StephanM87/Strenda-biocatalysis#57]
Dear all,
the current STRENDA modelling approach, as presented here in this repository appears to model the "biocatalysis world" in a rather rigid way, by specifying a lot of fine-grained details (e.g., vessels, mixing, operation mode, ...).
Since science, especially biocatalysis develops very rapidly, such a modelling attempts will quickly out-date and constant maintenance would be required. 
From our experience in standardisation lab-communication (SiLA) and lab data standard (AnIML), we learned that it is much more future prove to specify how something (in the current case **biocatalytic reactions**) should be described in a standardised way, rather than attempting to model the whole (biocatalysis) world. 
In both (SiLA and AnIML) the separation of core concepts and its implementation was very successful: [SiLA specification](https://sila2.gitlab.io/sila_base/)

Let me illustrate this by an example:

Rather then modelling a vessel, the mixing type and the hydrodynamics within this vessel etc. 
one could specify, how a reaction environment (which can be a vessel, but can be in future something completely different, like a sound elevated droplet  or whatever environment the future will bring) should be described:

```
Normative definition of the formal requirements:
C1: The reaction environment of a concrete biocataytic reaction shall be described by 
1. environment type
2. physical properties of this environment (e.g. hydrodynamic properties, heat transfer properties, mixing propteries)
3. all physical properties should be referenced by a well defined vocabulary, taxonomy or ontology
4. all physical properties shall have a reference to a unit. The recommended unit system is SI.
5. ....

Normative specification of the implementation (v1.0):
ImplC1: The data schema implementing C1 should be defined as an LinkML schema.
A recommended representation of an instance the reaction environment should be expressed as a JSON-LD file, 
which can be automatically generated from the LinkML schema.

Environment type should be specified, using the EnzymeXYOntology,
Physical Property should be specified, using the PropYZOntology,
.....

[ "@context" : "...",
  "ReactionEnvironment" : [ .... ],
  .....
]
```

Such an approach makes the whole concept future prove and maintains a flexible, versioned modelling, yet machine readable and actionable.

Regarding the granularity of the STRENDA data model,  a collection of [competency questions](https://www.sciencedirect.com/science/article/pii/S1570826819300617), collected before and during the standardisation process will determine how fine the terms need to be modeled and specified.

**Comment by rgiessmann** [comment]
Hi @markdoerr ,

while I believe that you are right about your approach, we wanted to start off with a very researcher-friendly (read: wet-lab experimentalist-friendly) means of putting things in biocatalysis into a more structured way.

You are probably true that our approach is not future-proof, especially if you consider the future to be fully ontology-driven and automated. We feel that we (the world) are not there yet, though. With our approach we want to inspire researchers to bring structure into their data at all, which we feel is the most important next step currently. 

I would like to postpone the discussion about the best modelling principle thus until we have gained more traction.

Thanks for your understanding! Best,
   Robert

**Comment by markdoerr** [comment]
Hi @rgiessmann,

any structure is (in most cases) better than no structure at all, I fully agree with you.
With our LARA suite we are experimenting and developing meta-structures, as I described above - and make them publicly available for the wet-lab people. So we can work a little bit in parallel - getting people structured in the first place, and giving them the right tool to make these structuring very easy (nearly naturally) by providing good meta-structures (= a structure describing how to structure things).

**Comment by markdoerr** [comment]
One very practical thing, we still should keep working on are the above mentioned "**competency questions**":
So general questions, we would would link to get answered, when we structure our wet-lab results.
These questions are very useful to check, if the modelling goes in the right direction.
To collect these questions, we could simply make a folder in this repository, called "competency_questions" and add a markdown file within, that could be even directly edited with the github web-interface (so no git-download=pull/push cycle  required).
